src/safeposix/syscalls/fs_calls.rs

```diff
@@ -26,33 +51,48 @@ impl Cage {
     }
 
     pub fn open_syscall(&self, path: &str, flags: i32, mode: u32) -> i32 {
-        //Check that path is not empty
+        // Check that the given input path is not empty
         if path.len() == 0 {
             return syscall_error(Errno::ENOENT, "open", "given path was null");
         }
+        
+        // Retrieve the absolute path from the root directory. The absolute path is then used to validate directory paths
+        // while navigating through subdirectories and creating a new file or open existing file at the given location.
```

Comment:
Does this also follow symlinks?

Reply:
I'll validate this with a test, but @rennergade can you please share your thoughts on this?

Reply:
We have no concept of symlinks so no